### ISSUE TITLE
Use polling instead of inotify for remote_syslog on RC containers

### ISSUE THREAD

**Issue by iprunache** [NRGI/resourcecontracts.org#1082]
## Why
`remote_syslog` exhausts all inotify instances available on the container instance and starts reporting that it cannot track new log files.

More details on issue NRGI/nrgi-dcos#240.

## What
- [x] `remote_syslog` is started with the `--poll` flag or `logrotate` is configured to use `copytruncate` on all RC containers
- [x] `remote_syslog` is started with the `--poll` flag or `logrotate` is configured to use `copytruncate` on all Shiny containers
- [x] document how to increase the limit of inotify watches on the container instance

## Notes
Papertrail recommends  to instruct logrotate to use copytruncate to rotate logs: https://github.com/papertrail/remote_syslog#log-rotation .

That will not be enough for all containers, though. The shiny server creates and deletes a lot of logs by itself so for it we may just have to use polling with `remote_syslog`.

Additionally, Docker itself seems to use a lot of inotify instances as shown in this comment: https://github.com/NRGI/resourcecontracts.org/issues/1082#issuecomment-350997450 . We should also look into increasing the inotify watches limit on the container instance.

We need to be careful of one thing with this: 3 of the 4 RC repos are on the YoungInnovations organization and all 4 repos were actual mirrors of GitLab repos hosted with YoungInnovations. They were periodically overwriting the GitHub repos with the content from the GitLab ones. If that's the case, we can't push the changes directly to GitHub as they would be overwritten, we'd have to push to GitLab. @manishgs can help us with this.

**Comment by iprunache** [comment]
@manishgs are the GitHub repos for RC still mirrors of your GitLab repos? We would need to push some code on all of them for this issue and wanted to make sure we do it the right way. Thanks!

**Comment by manishgs** [comment]
@iprunache 
We've discard mirroring gitlab-github repos so you can push to github.

**Comment by iprunache** [comment]
This also affects the Shiny containers since there are a lot of files created there too.

**Comment by iprunache** [comment]
A better solution may be to instruct `logrotate` to use `copytruncate` to rotate logs: https://github.com/papertrail/remote_syslog#log-rotation .

**Comment by iprunache** [comment]
This is becoming a bigger problem as we're deploying more containers to the same instance, and not just for RC. An example below.

List of processes using inotify instances:
```
[root@ip-10-0-1-17 ~]# for foo in /proc/*/fd/*; do readlink -f $foo; done | grep inotify | sort | uniq -c | sort -nr
     31 /proc/5951/fd/anon_inode:inotify
     25 /proc/3391/fd/anon_inode:inotify
     17 /proc/3191/fd/anon_inode:inotify
     17 /proc/20791/fd/anon_inode:inotify
     16 /proc/1379/fd/anon_inode:inotify
     15 /proc/14441/fd/anon_inode:inotify
     13 /proc/20660/fd/anon_inode:inotify
     12 /proc/9534/fd/anon_inode:inotify
     11 /proc/9973/fd/anon_inode:inotify
     10 /proc/3116/fd/anon_inode:inotify
      9 /proc/23994/fd/anon_inode:inotify
      6 /proc/2932/fd/anon_inode:inotify
      4 /proc/9441/fd/anon_inode:inotify
      4 /proc/24161/fd/anon_inode:inotify
      4 /proc/14538/fd/anon_inode:inotify
      2 /proc/5819/fd/anon_inode:inotify
      2 /proc/1/fd/anon_inode:inotify
      1 /proc/2893/fd/anon_inode:inotify
      1 /proc/24131/fd/anon_inode:inotify
      1 /proc/2411/fd/anon_inode:inotify
      1 /proc/1666/fd/anon_inode:inotify
      1 /proc/14488/fd/anon_inode:inotify
      1 /proc/1382/fd/anon_inode:inotify
      1 /proc/12243/fd/anon_inode:inotify
      1 /proc/12242/fd/anon_inode:inotify
```

And the actual processes:
```
[root@ip-10-0-1-17 ~]# ps 5951
  PID TTY      STAT   TIME COMMAND
 5951 ?        Sl     8:24 /usr/local/bin/remote_syslog -D
[root@ip-10-0-1-17 ~]# ps 3391
  PID TTY      STAT   TIME COMMAND
 3391 ?        Sl     5:20 /usr/local/bin/remote_syslog -D
[root@ip-10-0-1-17 ~]# ps 3191
  PID TTY      STAT   TIME COMMAND
 3191 ?        Sl   128:09 /usr/bin/dockerd --default-ulimit nofile=1024:4096 --storage-driver devicemapper --storage-opt dm.thinpooldev=/dev/mapper/docker-docker--pool --storage-opt dm.use_deferred_removal=true --storage-opt dm.
[root@ip-10-0-1-17 ~]# ps 20791
  PID TTY      STAT   TIME COMMAND
20791 ?        Sl     1:35 /usr/local/bin/remote_syslog -D
[root@ip-10-0-1-17 ~]# ps 1379
  PID TTY      STAT   TIME COMMAND
 1379 ?        Sl     0:00 /usr/bin/docker-proxy -proto tcp -host-ip 0.0.0.0 -host-port 32780 -container-ip 172.17.0.9 -container-port 80
[root@ip-10-0-1-17 ~]# ps 14441
  PID TTY      STAT   TIME COMMAND
14441 ?        Sl     0:00 /usr/bin/docker-proxy -proto tcp -host-ip 0.0.0.0 -host-port 32779 -container-ip 172.17.0.4 -container-port 3838
[root@ip-10-0-1-17 ~]# ps 20660
  PID TTY      STAT   TIME COMMAND
20660 ?        Sl     0:00 /usr/bin/docker-proxy -proto tcp -host-ip 0.0.0.0 -host-port 5020 -container-ip 172.17.0.8 -container-port 80
[root@ip-10-0-1-17 ~]# ps 9534
  PID TTY      STAT   TIME COMMAND
 9534 ?        Sl     4:03 /usr/local/bin/remote_syslog -D
```

We may have to increase the limit for inotify watches on the container instances in addition to optimizing `logrotate` and `remote_syslog` usage.

**Comment by iprunache** [comment]
Fixed this by using the just released v0.20 of `remote_syslog`.

https://github.com/papertrail/remote_syslog2/issues/212
